### PR TITLE
Parallel FD/CS

### PR DESCRIPTION
**Pull request by naylor-b**

### CODE REVIEW

File: openmdao/approximation_schemes/finite_difference.py [hunk 1]
```diff
@@ -228,34 +202,162 @@ def compute_approximations(self, system, jac=None, deriv_type='partial'):
...
+        total : bool
+            If True total derivatives are being approximated, else partials.
+
+        Returns
+        -------
+        ndarray
+            The results from running the perturbed system.
+        """
+        # TODO: MPI
+
```
Comment (Kenneth-T-Moore) [inline openmdao/approximation_schemes/finite_difference.py]: Is MPI still a TODO?

File: openmdao/approximation_schemes/complex_step.py [hunk 2]
```diff
@@ -77,38 +83,75 @@ def _key_fun(approx_tuple):
...
+    def compute_approximations(self, system, jac, total=False):
         """
         Execute the system to compute the approximate sub-Jacobians.
 
         Parameters
         ----------
         system : System
             System on which the execution is run.
-        jac : None or dict-like
+        jac : dict-like
```
[the remark this answers is not in the document]
Comment (naylor-b) [reply openmdao/approximation_schemes/complex_step.py]: fixed

File: openmdao/test_suite/components/cycle_comps.py [hunk 3]
```diff
@@ -92,9 +92,9 @@ def initialize(self):
         self.options.declare('connection_type', default='explicit',
                               values=['explicit', 'implicit'],
                               desc='How to connect variables.')
-        self.options.declare('finite_difference', default=False,
-                              types=bool,
-                              desc='If the derivatives should be finite differenced.')
+        self.options.declare('partial_method', default='exact',
+                              types=str,
+                              desc='How derivatives should be solves (exact, fd, or cs)')
```
[the remark this answers is not in the document]
Comment (naylor-b) [reply openmdao/test_suite/components/cycle_comps.py]: fixed

File: openmdao/test_suite/components/cycle_comps.py [hunk 4]
```diff
@@ -92,9 +92,9 @@ def initialize(self):
         self.options.declare('connection_type', default='explicit',
                               values=['explicit', 'implicit'],
                               desc='How to connect variables.')
-        self.options.declare('finite_difference', default=False,
-                              types=bool,
-                              desc='If the derivatives should be finite differenced.')
+        self.options.declare('partial_method', default='exact',
+                              types=str,
```
[the remark this answers is not in the document]
Comment (naylor-b) [reply openmdao/test_suite/components/cycle_comps.py]: fixed

File: openmdao/test_suite/groups/cycle_group.py [hunk 5]
```diff
@@ -63,9 +63,9 @@ def initialize(self):
         self.options.declare('partial_type', default='array',
                              values=['array', 'sparse', 'aij'],
                              desc='type of partial derivatives')
-        self.options.declare('finite_difference', default=False,
-                             types=bool,
-                             desc='If the derivatives should be finite differenced.')
+        self.options.declare('partial_method', default='exact',
+                             types=str,
```
[the remark this answers is not in the document]
Comment (naylor-b) [reply openmdao/test_suite/groups/cycle_group.py]: fixed

**Pull request merged**